core/src/main/java/com/adobe/aem/dot/common/FileResolver.java

```diff
@@ -113,7 +116,21 @@ public FileResolver(String basePath) {
       File[] files = directory.listFiles(fileFilter);
 
       if (files != null && files.length > 0) {
-        return Arrays.asList(files);
+        List<File> directoryFiles = new ArrayList<>();
+        for (File nextFile: files) {
+          String name = PathUtil.stripOffDrive(nextFile.getPath());
+          if (name.startsWith(currentWorkingDirectory)) {
+            name = name.substring(currentWorkingDirectory.length() + 1);
+          } else {
+            String target = PathUtil.stripLastPathElement(filePath);
+            name = PathUtil.appendPaths(target, nextFile.getName());
+          }
+          List<File> nextFiles = resolveFiles(name, currentWorkingDirectory);
+          if (!nextFiles.isEmpty()) {
+            directoryFiles.addAll(nextFiles);
+          }
+        }
+        return directoryFiles;
```

Comment:
Cannot just return `return Arrays.asList(files);` because some of those might be directories.  If `allowDirectoryPath` is true, we need to follow those too.